store/root/store.go

```diff
@@ -298,14 +299,28 @@ func (s *Store) Commit(cs *store.Changeset) ([]byte, error) {
 		s.logger.Debug("commit header and version mismatch", "header_height", s.commitHeader.Height, "version", version)
 	}
 
-	// commit SS
-	if err := s.stateStore.ApplyChangeset(version, cs); err != nil {
-		return nil, fmt.Errorf("failed to commit SS: %w", err)
-	}
+	eg := new(errgroup.Group)
+
+	// commit SS async
+	eg.Go(func() error {
```

Comment:
How does it work if it's still committing but the next block is getting State before the full commitment has completed

Reply:
So `Commit` itself isn't asynchronous. Rather, internally, committing SS and SC is asynchronous. In other words, we flush SS and SC in goroutines and then wait for both of them to finish before returning from `Commit`. We cannot begin to process another block until `Commit` returns.

Reply:
ah okay, thank you